apply_factors_wrangler.py

```diff
@@ -0,0 +1,222 @@
+        # Apply factors to non-responder data
+        #
+        # Join imputed data with current period responders
+
+        factors_dataframe,receipt_handle = get_from_sqs(queue_url)
+        #using assumption that factors df is granular- unaggregated data
+
+        # Reads in non responder data
+        non_responder_dataframe = pd.DataFrame(read_data_from_s3(bucket_name, non_responder_data_file))
+
+        # Read in previous period data for current period non-responders
+        prev_period_data = pd.DataFrame(read_data_from_s3(bucket_name, s3_file))
+
+        # Filter so we only have those that responded in prev
+        prev_period_data = prev_period_data[prev_period_data['response_type'] == 2]
+
+        question_columns = ['Q601_asphalting_sand', 'Q602_building_soft_sand', 'Q603_concreting_sand', 'Q604_bituminous_gravel', 'Q605_concreting_gravel', 'Q606_other_gravel', 'Q607_constructional_fill']
+        prev_question_columns = ['prev_Q601_asphalting_sand', 'prev_Q602_building_soft_sand', 'prev_Q603_concreting_sand', 'prev_Q604_bituminous_gravel', 'prev_Q605_concreting_gravel', 'prev_Q606_other_gravel', 'prev_Q607_constructional_fill', 'responder_id']
+
+        for question in question_columns:
+            prev_period_data = prev_period_data.rename(index=str, columns={question: 'prev_' + question})
+
+        # Join prev data so we have those who responded in prev but not in current
+
+        non_responder_dataframe = pd.merge(non_responder_dataframe, prev_period_data[prev_question_columns], on='responder_id')
+
+        # Merge the factors onto the non responders
+        non_responders_with_factors = pd.merge(non_responder_dataframe, factors_dataframe[['region', 'strata', 'imputation_factor_Q601_asphalting_sand', 'imputation_factor_Q602_building_soft_sand', 'imputation_factor_Q603_concreting_sand', 'imputation_factor_Q604_bituminous_gravel', 'imputation_factor_Q605_concreting_gravel', 'imputation_factor_Q606_other_gravel', 'imputation_factor_Q607_constructional_fill']], on=["region", "strata"], how='inner')
+        #Non responder data should now contain all previous values and 7 imp columns
+        imputed_data = lambda_client.invoke(FunctionName=method_name, Payload=non_responders_with_factors.to_json(orient='records'))
+
+        json_response = json.loads(imputed_data.get('Payload').read().decode("ascii"))
+
+        imputed_non_responders = pd.read_json(str(json_response).replace("'",'"'))
+        current_responders = factors_dataframe[factors_dataframe['period'] == int(current_period)]
+
+        final_imputed = pd.concat([current_responders, imputed_non_responders])
+
+        imputation_run_type = "Imputation complete"
+        message = final_imputed.to_json(orient='records')
+
+        send_output_to_sqs(queue_url,message,sqs_messageid_name,receipt_handle)
+
+        send_sns_message(arn, imputation_run_type, checkpoint)
+
+    except Exception as exc:
+        print(_get_traceback(exc))
```

Comment:
More test code? 

There's more print statements and queue purging I'm not sure should be here?